lib/libvfio-user.c

```diff
@@ -468,29 +468,39 @@ consume_fd(int *fds, size_t nr_fds, size_t index)
 
 int
 handle_dma_map(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
-               struct vfio_user_dma_region *region)
+               struct vfio_user_dma_map *region)
 {
     char rstr[1024];
     int fd = -1;
     int ret;
+    uint32_t prot = 0;
 
     assert(vfu_ctx != NULL);
     assert(msg != NULL);
     assert(region != NULL);
 
-    if (msg->in_size != sizeof(*region)) {
+    if (msg->in_size < sizeof(*region) || region->argsz < sizeof(*region)) {
         vfu_log(vfu_ctx, LOG_ERR, "bad size of DMA map region %zu",
```

Comment:
probably need to report argsz in the error message now too

Reply:
ok